### PR TITLE
[ci][image_picker]enable xcode 12/iOS 14 for all tasks except lint

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

This PR moves all the iOS tasks to XCode 12 and iOS 14.
Note: Pod linter task could not move to XCode 12 yet because flutter engine doesn't support arm simulators yet. So I created a separate task with XCode 11 for pod lint task.

This also fixes some image picker xctest so that it works on the new ci env.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by fkorotkov** [comment]
@cyanglaz, are you planning to continue working on this issue? FYI `big-sur-xcode-12.3` image is also available.

**Comment by cyanglaz** [comment]
@fkorotkov Yes. I saw your announcement and I'll make this a priority now.

**Comment by fkorotkov** [comment]
@cyanglaz thank you! Seems only `camera` plugin is failing with Xcode 12. I tested it tomorrow here: https://cirrus-ci.com/task/6096381335044096

**Comment by cyanglaz** [comment]
@fkorotkov The last time I tried, the image picker xcode test was constantly failing and I couldn't reproduce the failure locally with the same setup. Did anything change? Do I need to use a different mac/xcode config than it is in this PR?

**Comment by fkorotkov** [comment]
I tried 12.0 and you tried 12.1. Do you have a preference of the Xcode? Or for example the latest big-sur-xcode-12.3 will work for you? Or you want to use older version for backward compatibility?

**Comment by cyanglaz** [comment]
@fkorotkov tried 12.0 didn't work. is it big-sur-xcode-12.3-flutter`?

**Comment by fkorotkov** [comment]
It's `big-sur-xcode-12.3` Starting with Big Sur we removed `-flutter` images so the Xcode one includes Flutter now. See https://cirrus-ci.org/guide/macOS/?h=+macos#macos-big-sur

**Comment by cyanglaz** [comment]
@fkorotkov Looks like there are still some issue with the image picker controller UI on ci. It's failing constantly and I can't repro it locally. :(

**Comment by fkorotkov** [comment]
@cyanglaz how are you trying to reproduce it locally? Do you also create an iPhone 11 simulator with the same OS version?

**Comment by cyanglaz** [comment]
@fkorotkov I did it with the same xcode version and same simulator. The OS version on my machine: 10.15.7 (19H114). I am not sure if the OS version would affect how the image picker controller perform on simulators?

**Comment by fkorotkov** [comment]
@cyanglaz interesting! I tried to run your branch on the new infrastructure by adding `experimental: true` to the config and everything worked https://cirrus-ci.com/build/6518411029118976. Do you want to switch to the new infrastructure earlier?)

**Comment by fkorotkov** [comment]
Also note how much faster the tasks are! 😍

**Comment by fkorotkov** [comment]
Hm. Same error on the new virtualization. I wonder why it worker (or was it ignored on my branch?).

**Comment by cyanglaz** [comment]
@fkorotkov Yeah looks like it was ignored.

**Comment by fkorotkov** [comment]
It seems iOS version did change switching to Xcode 12 which means the test follows another code path here https://github.com/flutter/plugins/blob/07cf89a08b8800f3337049654bc5c1c19edd50d0/packages/image_picker/image_picker/example/ios/RunnerUITests/ImagePickerFromGalleryUITests.m#L165

Is it possible that the error in the test? Do you have a iOS 14 device locally?

**Comment by cyanglaz** [comment]
The log says it went to iOS 14 (the test was looking for Image which is iOS 14 only) which is correct.
I think the failure is caused due to a lost connection to the app. 

looking at here:
```
Waiting 30.0s for Button to exist
    t =     9.57s     Checking `Expect predicate `exists == 1` for object Button`
    t =     9.57s         Checking existence of `Button`
    t =     9.59s         Capturing element debug description
    t =    10.65s     Checking `Expect predicate `exists == 1` for object Button`
    t =    10.65s         Checking existence of `Button`
    t =    10.67s         Capturing element debug description
    t =    11.73s     Checking `Expect predicate `exists == 1` for object Button`
    t =    11.73s         Checking existence of `Button`
    t =    11.90s Checking existence of `Button`
    t =    11.96s Tap Button
    t =    11.96s     Wait for io.flutter.plugins.imagePickerExample to idle
    t =    12.01s     Find the Button
    t =    12.06s     Check for interrupting elements affecting "Cancel" Button
    t =    12.11s     Synthesize event
    t =    12.22s     Wait for io.flutter.plugins.imagePickerExample to idle
```

The first time, bringing up the image picker was fine, and the cancel button was successfully clicked, the image picker controller was successfully dismissed. But the second time, it was crashing when looking for an image.

**Comment by cyanglaz** [comment]
Ok, it looks like it's always crashing at the second time when image picker is brought up. pretty similar to https://github.com/flutter/flutter/issues/67430
But i thought this issue should be resolved on metal simulators. I can't repro this locally either.

**Comment by cyanglaz** [comment]
Looks like bringing up a new App for each test fixed the issue on CI. I'm still not sure what's going on with bringing up 2 consecutive image picker on ci. I think for now we should make each test session separate so we can enable xcode 12 first. Then figure out what's going on with the particular image picker tests.

**Comment by fkorotkov** [comment]
As far as I know neither Anka nor the new Parallels virtualization do support Metal inside VMs. They use Metal externally but don't support "nested Metal".

File: .cirrus.yml [hunk 1]
```diff
@@ -134,13 +134,14 @@ task:
         - xvfb-run ./script/incremental_build.sh drive-examples --linux
 
 task:
+  # XCode 11 task
```
Comment (jmagman) [inline .cirrus.yml:137]: nit: casing is `Xcode`.

File: .cirrus.yml [hunk 2]
```diff
@@ -134,13 +134,14 @@ task:
...
+  # TODO(cyanglaz): merge XCode 11 task to XCode 12 task when all the matrix can be run in XCode 12.
   # don't run on release tags since it creates O(n^2) tasks where n is the number of plugins
   only_if: $CIRRUS_TAG == ''
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
   osx_instance:
     image: catalina-xcode-11.3.1-flutter
   upgrade_script:
-    - sudo gem install cocoapods
```
Comment (jmagman) [inline .cirrus.yml:143]: Should we keep this to keep CocoaPods up to date?

File: packages/image_picker/image_picker/example/ios/RunnerUITests/ImagePickerFromGalleryUITests.m [hunk 3]
```diff
@@ -53,11 +53,18 @@ - (void)setUp {
                                         }];
 }
 
+- (void)tearDown {
+  [self.app terminate];
```
Comment (jmagman) [inline packages/image_picker/image_picker/example/ios/RunnerUITests/ImagePickerFromGalleryUITests.m:57]: -[super testDown];

**Review by jmagman — APPROVED** [review]
LGTM with nits if it works!

**Comment by cyanglaz** [comment]
@jmagman Thanks for the quick review! Fixed!

**Pull request merged**